lib/generators/deimos/schema_class_generator.rb

```diff
@@ -122,7 +122,13 @@ def write_file(schema, key_config)
 
           file_prefix = schema.name.underscore.singularize
           if Deimos.config.schema.generate_namespace_folders
-            file_prefix = "#{@modules.last.underscore.singularize}/#{file_prefix}"
+            # Use entire namespace for folders
+            # but don't add directories that are already in the path
+            directories = @modules.map { |m|
+              Deimos.config.schema.generated_class_path.include?(m.underscore) ? nil : m.underscore
+            }.compact
```

Comment:
Wouldn't this be clearer as
`directories = @modules.map(&:underscore).select { |m| Deimos.config.schema.generated_class_path.exclude?(m) }`

Reply:
yes thx, will update